contracts/InterchainTokenService.sol

```diff
@@ -785,68 +785,30 @@ contract InterchainTokenService is
         IGatewayCaller.MetadataVersion metadataVersion,
         uint256 gasValue
     ) internal {
-        string memory destinationAddress;
+        if (!isTrustedChain(destinationChain)) revert UntrustedChain();
 
-        (destinationChain, destinationAddress, payload) = _getCallParams(destinationChain, payload);
+        payload = abi.encode(MESSAGE_TYPE_SEND_TO_HUB, destinationChain, payload);
+        destinationChain = ITS_HUB_CHAIN_NAME;
 
-        _callContract(destinationChain, destinationAddress, payload, metadataVersion, gasValue);
+        _callContract(payload, metadataVersion, gasValue);
     }
 
     /**
      * @notice Calls a contract on a destination chain via the gateway caller.
-     * @param destinationChain The chain where the contract will be called.
-     * @param destinationAddress The address of the contract to call.
      * @param payload The data payload for the transaction.
      * @param metadataVersion The version of the metadata.
      * @param gasValue The amount of gas to be paid for the transaction.
      */
-    function _callContract(
-        string memory destinationChain,
-        string memory destinationAddress,
-        bytes memory payload,
-        IGatewayCaller.MetadataVersion metadataVersion,
-        uint256 gasValue
-    ) internal {
-        // Check whether no trusted address was set for the destination chain
-        if (bytes(destinationAddress).length == 0) revert UntrustedChain();
+    function _callContract(bytes memory payload, IGatewayCaller.MetadataVersion metadataVersion, uint256 gasValue) internal {
+        string memory hubAddress = itsHubAddress();
 
         (bool success, bytes memory returnData) = gatewayCaller.delegatecall(
-            abi.encodeWithSelector(
-                IGatewayCaller.callContract.selector,
-                destinationChain,
-                destinationAddress,
-                payload,
-                metadataVersion,
-                gasValue
-            )
+            abi.encodeWithSelector(IGatewayCaller.callContract.selector, ITS_HUB_CHAIN_NAME, hubAddress, payload, metadataVersion, gasValue)
```

Comment:
Maybe in a follow up PR, we could remove the gateway caller util? We don't have gateway token support anymore, so the extra gas cost and complexity seems unnecessary.
https://axelarnetwork.atlassian.net/browse/AXE-8398

Reply:
The gateway caller serves mainly to reduce contract size, but maybe now we can fit it into ITS, worth trying but unlikely.